apps/typeform/frontend/src/Auth/TypeformOAuth.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect } from 'react';
+import { AppExtensionSDK } from 'contentful-ui-extensions-sdk';
+import { Button } from '@contentful/forma-36-react-components';
+
+interface Props {
+  sdk: AppExtensionSDK;
+  expireSoon?: boolean;
+  setToken: (token: string) => void;
+}
+
+export function TypeformOAuth({ sdk, expireSoon, setToken, ...rest }: Props) {
+  let oauthWindow: Window | null;
+
+  useEffect(() => {
+    sdk.app?.setReady();
+
+    return () => window.removeEventListener('message', handleTokenEvent);
+  }, []);
+
+  const executeOauth = () => {
+    const url = `https://api.typeform.com/oauth/authorize?&client_id=${
+      process.env.CLIENT_ID
+    }&redirect_uri=${encodeURIComponent(
+      process.env.OAUTH_REDIRECT_URI as string
+    )}&scope=forms:read+workspaces:read&state=${encodeURIComponent(window.location.href)}`;
+
+    oauthWindow = window.open(url, 'Typeform Contentful', 'left=150,top=10,width=800,height=900');
+
+    window.addEventListener('message', handleTokenEvent);
+  };
+
+  const handleTokenEvent = ({ data }: any) => {
+    const { token, error } = data;
+
+    if (error) {
+      console.error('There was an error authenticating. Please refresh and try again.');
```

Comment:
If there is an error, shouldn't we close the window and have that error show up somewhere in the webapp UI? I think leaving the window open won't really help much.